### PR TITLE
Customer request: Add support for clearing the spatial observations

### PR DESCRIPTION
**Pull request by davidkline-ms**
This feature was requested by a customer in our online forums.

This change adds the ability to request the spatial observations be cleared (removed from the scene and freed). 

It also renames SpatialAwarenessObjParent to SpatialAwarenessObserverParent for clarity.

The Spatial Awareness demo scene has been updated to add a sphere that allows for toggling the observers (first click suspends and clears, second click resumes, etc).

Testing performed
- Build and deploy to HoloLens (il2cpp and .net backends). Confirm that the meshes clear and resume.
- Build and run in editor using an early file observer prototype.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs [hunk 1]
```diff
@@ -170,18 +170,27 @@ public override void Destroy()
         private GameObject spatialAwarenessObjectParent = null;
 
         /// <inheritdoc />
-        public GameObject SpatialAwarenessObjectParent => spatialAwarenessObjectParent != null ? spatialAwarenessObjectParent : (spatialAwarenessObjectParent = CreateSpatialAwarenessParent);
+        public GameObject SpatialAwarenessObjectParent => spatialAwarenessObjectParent != null ? spatialAwarenessObjectParent : (spatialAwarenessObjectParent = CreateSpatialAwarenessObjectParent);
```
Comment (lukastoenneMS) [inline Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs]: Are the renamings of SpatialAwarenessObjectParent and CreateSpatialAwarenessObjectParent the only breaking changes?

File: Assets/MixedRealityToolkit.Examples/Demos/SpatialAwareness/Scripts/ClearSpatialObservations.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        /// </summary>
+        private bool clearObservations = true;
+
+        private void Start()
+        { }
+
+        /// <summary>
+        /// Updates the state of the observers.
+        /// </summary>
+        public void UpdateObservers()
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/SpatialAwareness/Scripts/ClearSpatialObservations.cs]: nit:  i would call this "ToggleObservers" / "ToggleObserverActive" or "ToggleObserverState"- I wouldn't expect an update method to change the state of the observer

**Comment by thalbern** [comment]
would be great if we could add tests for this: creating the spatial observer, suspending, clearing, resuming

File: Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs]: No, the addition of the ClearObservation methods is also a breaking change

**Comment by davidkline-ms** [comment]
> would be great if we could add tests for this: creating the spatial observer, suspending, clearing, resuming

Once i get the file / model observer PR completed, i can add a test for it. :)

File: Assets/MixedRealityToolkit.Examples/Demos/SpatialAwareness/Scripts/ClearSpatialObservations.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Examples/Demos/SpatialAwareness/Scripts/ClearSpatialObservations.cs]: done :)

**Pull request merged**